src/main/java/jcifs/smb1/smb1/TransactNamedPipeInputStream.java

```diff
@@ -0,0 +1,136 @@
+/* jcifs smb client library in Java
+ * Copyright (C) 2000  "Michael B. Allen" <jcifs at samba dot org>
+ * 
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * 
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+package jcifs.smb1.smb1;
+
+import java.io.InputStream;
+import java.io.IOException;
+import java.net.UnknownHostException;
+import java.net.MalformedURLException;
+
+class TransactNamedPipeInputStream extends SmbFileInputStream {
+
+    private static final int INIT_PIPE_SIZE = 4096;
+
+    private byte[] pipe_buf = new byte[INIT_PIPE_SIZE];
+    private int beg_idx, nxt_idx, used;
+    private boolean dcePipe;
+
+    Object lock;
+
+    TransactNamedPipeInputStream( SmbNamedPipe pipe ) throws SmbException,
+                MalformedURLException, UnknownHostException {
+        super( pipe, ( pipe.pipeType & 0xFFFF00FF ) | SmbFile.O_EXCL );
+        this.dcePipe = ( pipe.pipeType & SmbNamedPipe.PIPE_TYPE_DCE_TRANSACT ) != SmbNamedPipe.PIPE_TYPE_DCE_TRANSACT;
+        lock = new Object();
+    }
+    public int read() throws IOException {
+        int result = -1;
+
+        synchronized( lock ) {
+            try {
+                while( used == 0 ) {
+                    lock.wait();
+                }
+            } catch( InterruptedException ie ) {
+                throw new IOException( ie.getMessage() );
+            }
+            result = pipe_buf[beg_idx] & 0xFF;
+            beg_idx = ( beg_idx + 1 ) % pipe_buf.length;
+        }
+        return result;
+    }
+    public int read( byte[] b ) throws IOException {
+        return read( b, 0, b.length );
+    }
+    public int read( byte[] b, int off, int len ) throws IOException {
+        int result = -1;
+        int i;
+
+        if( len <= 0 ) {
+            return 0;
+        }
+        synchronized( lock ) {
+            try {
+                while( used == 0 ) {
+                    lock.wait();
+                }
+            } catch( InterruptedException ie ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)